### ISSUE TITLE
Slow PDF Reading Times

### ISSUE THREAD

**Issue by malekinho8** [whitead/paper-qa#68]
Using `pypdf` to read PDF files when creating the context chunks is orders of magnitude slower than packages such as `fitz` and `pdftotext`. The `fitz` package cannot be used due to licensing conflicts, however, `pdftotext` is under the MIT license, and may be a potential solution.

**Comment by malekinho8** [comment]
Regarding this topic, I performed a basic preliminary experiment to calculate time to read PDFs between different PDF reading libraries. The PDF I was experimenting with was a standard scientific article ~8 pages in length. The results can be seen [here](https://user-images.githubusercontent.com/78977545/232334675-4db50be3-aad7-4784-8607-37e3f0a07fa6.png). I've also included the length of the extracted text for each test to show that some extraction methods incur some loss even though they may have shorter run times. Still, `fitz` and `pdftotext` offer the best in terms of run time and overall text extraction.

The code used to generate these results is pasted below:

```python
from pdfminer.high_level import extract_text
from PyPDF4 import PdfFileReader
import fitz
import pdftotext
import time

pdf_file_name = "/Users/malek8/Zotero/storage/GLMPHG53/Beaulieu et al. - 2007 - A complete analysis of the laser beam deflection s.pdf"
run_pdfminer = True

if run_pdfminer:
    # calculate time of extraction between two methods
    time1 = time.time()
    text = extract_text(pdf_file_name)
    time2 = time.time()

    # print the time and label
    print("Time taken by pdfminer: ", time2 - time1, " seconds")
    print("Length of text: ", len(text))
    print("\n\n")

# use fitz to load PDF and extract text
num_pages = 0
time1 = time.time()
doc = fitz.open(pdf_file_name)
text = ""
for i, page in enumerate(doc):
    text += doc.get_page_text(i)
    num_pages += 1
time2 = time.time()

# print the time and label, and length of text string
print("Time taken by PyMuPDF: ", time2 - time1, " seconds")
print("Length of text: ", len(text))
print("\n\n")

# do the same for PyPDF4
time1 = time.time()
with open(pdf_file_name, "rb") as f:
    pdf = PdfFileReader(f)
    text = ""
    for i in range(pdf.numPages):
        text += pdf.getPage(i).extractText()
time2 = time.time()

# print the time and label
print("Time taken by PyPDF4: ", time2 - time1, " seconds")
print("Length of text: ", len(text))
print("\n\n")

# do the same for pdftotext
time1 = time.time()
with open(pdf_file_name, "rb") as f:
    pdf = pdftotext.PDF(f)
    text = ""
    for page in pdf:
        text += page
time2 = time.time()

# print the time and label
print("Time taken by pdftotext: ", time2 - time1, " seconds")
print("Length of text: ", len(text))
print("\n\n")

# print the number of pages and label
print("Number of pages: ", num_pages)
```

**Comment by whitead** [comment]
Fixed by #126
